src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -113,177 +138,319 @@ protected AbstractApiAction(
+            return ValidationResult.success(securityConfiguration);
+        }).orElse(ValidationResult.success(securityConfiguration)));
+    }
 
-    protected void handleApiRequest(final RestChannel channel, final RestRequest request, final Client client) throws IOException {
+    protected final ValidationResult<SecurityConfiguration> processPatchRequest(final RestRequest request) throws IOException {
+        return loadConfiguration(nameParam(request), false).map(
+            securityConfiguration -> withPatchRequestContent(request).map(
+                patchContent -> securityConfiguration.maybeEntityName()
+                    .map(entityName -> patchEntity(request, patchContent, securityConfiguration))
+                    .orElseGet(() -> patchEntities(request, patchContent, securityConfiguration))
+            )
+        );
+    }
 
+    protected final ValidationResult<JsonNode> withPatchRequestContent(final RestRequest request) {
         try {
-            switch (request.method()) {
-                case DELETE:
-                    handleDelete(channel, request, client, null);
-                    break;
-                case POST:
-                    createValidator().validate(request)
-                        .valid(jsonContent -> handlePost(channel, request, client, jsonContent))
-                        .error(toXContent -> requestContentInvalid(request, channel, toXContent));
-                    break;
-                case PUT:
-                    createValidator().validate(request)
-                        .valid(jsonContent -> handlePut(channel, request, client, jsonContent))
-                        .error(toXContent -> requestContentInvalid(request, channel, toXContent));
-                    break;
-                case GET:
-                    handleGet(channel, request, client, null);
-                    break;
-                default:
-                    throw new IllegalArgumentException(request.method() + " not supported");
+            final var parsedPatchRequestContent = Utils.toJsonNode(request.content().utf8ToString());
+            if (!(parsedPatchRequestContent instanceof ArrayNode)) {
+                return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("Wrong request body"));
+            }
+            final var operations = patchOperations(parsedPatchRequestContent);
+            if (operations.isEmpty()) {
+                return ValidationResult.error(RestStatus.BAD_REQUEST, badRequestMessage("Wrong request body"));
             }
-        } catch (JsonMappingException jme) {
-            throw jme;
-            // TODO strip source
-            // if(jme.getLocation() == null || jme.getLocation().getSourceRef() == null) {
-            // throw jme;
-            // } else throw new JsonMappingException(null, jme.getMessage());
+            for (final var patchOperation : operations) {
+                if (!PATCH_OPERATIONS.contains(patchOperation)) {
```

Comment:
Curious, should this be case insensitive? What is the current behavior?

Reply:
This is a good question. I use upper case for `final static` variables and it is a habit nothing more. Out check style tells nothing about it. I can change it to `PATCH_OPERATIONS` as `patchOpertaions`.

Reply:
@cwperks Looks like we use default rules about naming in case of final static:
https://github.com/opensearch-project/security/blob/32d3112ad3415243572928a70e64e382cbdfd43b/checkstyle/sun_checks.xml#L106C40-L106C40

which is: 
```java
Code Example:

 class MyClass {
   public static int goodStatic = 2; // OK
   private static int BadStatic = 2; // violation, name 'BadStatic'
                                     // must match pattern '^[a-z][a-zA-Z0-9]*$'
 }
```
Wondering why it passes check style :-)